pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -225,18 +189,22 @@ func buildLogsQuery(start, end, step int64, mq *v3.BuilderQuery, fields map[stri
 			"from signoz_logs.distributed_logs " +
 			"where " + timeFilter + "%s " +
 			"group by %s%s " +
-			"order by %sts"
+			"order by %s"
 
 	groupBy := groupByAttributeKeyTags(mq.GroupBy...)
-	orderBy := orderByAttributeKeyTags(mq.OrderBy, mq.GroupBy)
+	orderBy := orderByAttributeKeyTags(panelType, mq.OrderBy, mq.GroupBy)
+
+	if mq.AggregateOperator != v3.AggregateOperatorNoOp {
+		if !strings.Contains(orderBy, "ts") {
+			orderBy = orderBy + "ts"
+		}
+		orderBy = strings.Replace(orderBy, "timestamp", "ts", 1)
```

Comment:
I would avoid the `.Contains` and `.Replace`. It's easy to have a case where this will break. Will this still work if there is a different order by could that contains substring `ts`, for instance, `requests_order` or similar? Will this still work if a column appears at the beginning of orderBy, which contains `timestamp` as a substring?

Reply:
Makes sense, have removed this logic